concent_api/gatekeeper/views.py

```diff
@@ -124,73 +161,193 @@ def parse_headers(request: WSGIRequest, path_to_file: str) -> Union[FileTransfer
+            path_to_file,
+            loaded_golem_message.subtask_id,
+            client_public_key,
+        )
     url_validator = URLValidator()
     try:
         url_validator(loaded_golem_message.storage_cluster_address)
     except ValidationError:
-        return gatekeeper_access_denied_response('storage_cluster_address is not a valid URL.', path_to_file, loaded_golem_message.subtask_id, client_public_key)
+        return gatekeeper_access_denied_response(
+            'storage_cluster_address is not a valid URL.',
+            ErrorCode.MESSAGE_STORAGE_CLUSTER_INVALID_URL.value,
+            path_to_file,
+            loaded_golem_message.subtask_id,
+            client_public_key,
+        )
     if loaded_golem_message.storage_cluster_address != settings.STORAGE_CLUSTER_ADDRESS:
-        return gatekeeper_access_denied_response('Given storage_cluster_address is not defined in settings.', path_to_file, loaded_golem_message.subtask_id, client_public_key)
+        return gatekeeper_access_denied_response(
+            'Given storage_cluster_address is not defined in settings.',
+            ErrorCode.MESSAGE_STORAGE_CLUSTER_UNKNOWN.value,
+            path_to_file,
+            loaded_golem_message.subtask_id,
+            client_public_key,
+        )
 
     # -CLIENT_PUBLIC_KEY
     if not isinstance(loaded_golem_message.authorized_client_public_key, bytes):
-        return gatekeeper_access_denied_response('Wrong type of authorized_client_public_key variable.', path_to_file, loaded_golem_message.subtask_id, client_public_key)
+        return gatekeeper_access_denied_response(
+            'Wrong type of authorized_client_public_key variable.',
+            ErrorCode.MESSAGE_AUTHORIZED_CLIENT_PUBLIC_KEY_WRONG_TYPE.value,
+            path_to_file,
+            loaded_golem_message.subtask_id,
+            client_public_key,
+        )
     client_public_key_base64 = (b64encode(loaded_golem_message.authorized_client_public_key)).decode('ascii')
     if client_public_key != client_public_key_base64:
-        return gatekeeper_access_denied_response('authorized_client_public_key is different then Concent-Client-Public-Key header.', path_to_file, loaded_golem_message.subtask_id, client_public_key)
+        return gatekeeper_access_denied_response(
+            'authorized_client_public_key is different then Concent-Client-Public-Key header.',
+            ErrorCode.MESSAGE_AUTHORIZED_CLIENT_PUBLIC_KEY_INVALID.value,
+            path_to_file,
+            loaded_golem_message.subtask_id,
+            client_public_key,
+        )
 
     # -OPERATION
     if request.method == 'POST' and loaded_golem_message.operation != 'upload':
-        return gatekeeper_access_denied_response('Wrong operation variable for this request method.', path_to_file, loaded_golem_message.subtask_id, client_public_key)
+        return gatekeeper_access_denied_response(
+            'Wrong operation value for this request method.',
```

Comment:
`Wrong operation value for this request method` -> `Upload requests must use POST method`.

Reply:
:+1: